src/utils/url-utils.ts

```diff
@@ -0,0 +1,102 @@
+import { useHistory } from 'react-router-dom';
+import moment from 'moment';
+
+export type URLParams = {
+  hazardLayerId?: string;
+  baselineLayerId?: string;
+  date?: number;
+};
+
+/*
+  transforms the string value according to the field specified in the
+  type URLParams
+*/
+const parseValue = (key: string, value: string): URLParams[keyof URLParams] => {
+  switch (key) {
+    case 'date':
+      return moment(value).valueOf();
+    default:
+      return value;
+  }
+};
+
+/*
+  Serialized value to string depending on the parameter specified in the
+  type URLParams
+*/
+const valueToString = (
+  key: string,
+  value: URLParams[keyof URLParams],
+): string => {
+  switch (key) {
+    case 'date':
+      return moment(value).format('YYYY-MM-DD');
+    default:
+      return value as string;
+  }
+};
+
+/*
+  Serialized value to string depending on the parameter specified in the
+  type URLParams
+*/
+const parseHistory = (locationSearch: string): URLParams =>
+  locationSearch === ''
+    ? ({} as URLParams)
+    : locationSearch
+        .substring(1)
+        .split('&')
+        .map(l => l.split('='))
+        .reduce(
+          (obj, [key, value]) => ({
+            ...obj,
+            [key]: parseValue(key, value),
+          }),
+          {} as URLParams,
+        );
+
+/*
+  Serialize urlParams object to string.
+*/
+const urlParamsToString = (params: URLParams): string =>
+  Object.entries(params)
+    .map(([key, value]) => `${key}=${valueToString(key, value)}`)
+    .join('&');
+
+/*
+  This custom hook tracks the browser url string, which is defined by the useHistory hook.
+  We created additional functions to update the url based on user events, such as select date
+  or select layer.
+*/
+export const useUrlHistory = () => {
+  const { replace, location } = useHistory();
```

Comment:
This entire function can be simplified quite a lot by using existing tools like URLSearchParams. Better to use existing infrastructure so we can guarantee it works without bugs!